contributing.md

```diff
@@ -128,11 +187,22 @@ To check that the package is compatible with server side rendering, build the pr
 yarn check:ssr
 ```
 
-### How to make an entropy source
+### How to add an entropy source
 
-Entropy source is a function that gets a piece of data about the browser a.k.a. a fingerprint component.
+An entropy source is a function that gets a piece of data about the browser.
+The value returned by an entropy source is used to produce the visitor identifier and called an entropy component.
 Entropy sources are located in the [src/sources](src/sources) directory.
-Entropy component must be a simple JS value that can be JSON-encoded.
+All entropy components must be simple JavaScript values that can be encoded into JSON.
+
+Entropy source must meet the following requirements:
+- It is stable, that is always or almost always produces the same value in each browser, including incognito, guest and desktop modes;
```

Comment:
I am not sure this "that is" phrasing is grammatically correct, at least I bumped on it. Also missing oxford comma. 


```suggestion
- It is stable — it always or almost always produces the same value in each browser, including incognito, guest, and desktop modes;
```

Reply:
Oxford comma is optional (hence the name), isn't it?

Reply:
Looks like it inclines to be required in American English, and we use it in other documents, so let's add it